server/resources_api/users.py

```diff
@@ -112,22 +115,27 @@ def post(self):
     def patch(self):
         try:
             args = user_update_args.parse_args()
-            user_id = args['user_id']
+            user_id = args["user_id"]
             user = db.session.query(UserTable).filter_by(user_id=user_id).first()
             # Update the user attributes if they are present in the args
-            if 'username' in args and args['username'] != None:
-                user.username = args['username']
-            if 'pwd' in args and args['pwd'] != None:
+            if "username" in args and args["username"] is not None:
+                user.username = args["username"]
+            if "pwd" in args and args["pwd"] is not None:
```

Comment:
The backend only checks for None. Maybe add a check for empty string. Or do not send empty strings in the patch request.